### ISSUE TITLE
Configurable Livebook URL

### ISSUE THREAD

**Issue by epinault** [elixir-lang/ex_doc#1428]
Would it be possilbe to make this https://github.com/elixir-lang/ex_doc/pull/1408/files#diff-2de343f41e8c9bc4a79c58c52c6170765675f8b437643ca73b7813911a23b97eR44 configurable and point to localhost when you have a local one running? The reason is we have private docs for private packages and being able to go directly to localhost would be better or any DNS without having to go to livebook.dev ?

**Comment by jonatanklosko** [comment]
livebook.dev doesn't run the notebook though, you only tell it where to redirect, in this case localhost. Pointing to a specific address would be brittle, especially that people may run Livebook on different ports.

**Comment by epinault** [comment]
yes, I get that , let me rephrase that

I , as doc publisher, know my doc viewer are all internal dev on private packages we have (and private docs) server. I already know they would need to use a localhost or our own hosted livebook server anyway. So having a way to override the default would allow me to provide a more direct experience for that case?

**Comment by jonatanklosko** [comment]
As soon as you have a hosted Livebook, people will have two ways of running the given notebook though, right? And my point is that even assuming localhost, someone may very well run Livebook on a different port.

Is your concern related to the additional click, or something else?

**Comment by epinault** [comment]
1. Additional click
2. Exposing inernal url and internal name (security team might get on my case) in the outside world

And yes, they could run a different port, but that on the dev to follow our prescription and tooling in our company.  

Dont get me wrong, it s awesome to have it for public notebook etc and perfect default behavior. Just for the above reason, giving a way to override sometime might be needed

**Comment by jonatanklosko** [comment]
I see. So the additional click is something we could make configurable on livebook.dev side, but the security concern is something else. @josevalim thoughts?

**Comment by josevalim** [comment]
One option I can think is to configure the ExDoc Livebook URL in the browser and we obey that. Because my concern with allowing a custom URL is that now someone can change their package on Hexdocs to point to something else that can be used to confuse the user.

**Comment by jonatanklosko** [comment]
@josevalim so we would have a UI where the user either picks livebook.dev or a "direct" redirect and specifies this instance? Any ideas where would that go? Maybe we should have a settings modal and move "toggle night mode" and "enable/disable tooltips" there too.

**Comment by josevalim** [comment]
@jonatanklosko yes, that's probably the best call.
